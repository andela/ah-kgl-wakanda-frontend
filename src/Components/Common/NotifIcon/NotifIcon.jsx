import React from 'react';
import { PropTypes } from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBell } from '@fortawesome/free-solid-svg-icons';
import './NotifIcon.scss';

/**
 * Notification Icon component
 * @description render the icon and badge for the number of notification
 * @param {object} props {icon, notificationsCount, onClick}
 * @returns {object} jsx
 */
const NotifIcon = ({ icon, notificationsCount, onClick }) => (
  <div className="user mr-3">
    <div role="button" tabIndex="-1" onKeyDown={onClick} className="drop-down" onClick={onClick}>
      <FontAwesomeIcon icon={icon} className="notif-icon" />
      {notificationsCount >= 1 ? <span className="num">{notificationsCount}</span> : null}
    </div>
  </div>
);

NotifIcon.propTypes = {
  icon: PropTypes.any,
  notificationsCount: PropTypes.number,
  onClick: PropTypes.func.isRequired,
};

NotifIcon.defaultProps = {
  icon: faBell,
  notificationsCount: 0,
};

export default NotifIcon;
