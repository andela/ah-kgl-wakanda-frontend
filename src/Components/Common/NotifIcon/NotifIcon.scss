@import '../../../assets/sass/components';
.user {
  display: flex;
  align-self: center;
  align-items: center;
  margin: 0 20px;
  .drop-down {
    .notif-icon {
      margin-top: auto;
      margin-left: 15px;
      font-size: 20px;
      color: $color-dark-grey;
    }

    .num {
      position: relative;
      bottom: 10px;
      right: 2px;
      padding: 0.2px 5px;
      border-radius: 50%;
      background: #f46036;
      color: $color-light;
      line-height: 18px;
      font-family: sans-serif;
      text-align: center;
    }
  }
}
