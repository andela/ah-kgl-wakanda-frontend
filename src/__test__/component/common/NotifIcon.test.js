import React from 'react';
import { shallow } from 'enzyme';
import NotifIcon from '../../../Components/Common/NotifIcon/NotifIcon';

describe('Logo component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      onClick: jest.fn(),
      notificationsCount: 2,
    };
    wrapper = shallow(<NotifIcon {...props} />);
  });

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
