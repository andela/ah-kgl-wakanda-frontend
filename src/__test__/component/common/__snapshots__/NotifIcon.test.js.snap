// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should render without crashing 1`] = `ShallowWrapper {}`;
